### ISSUE TITLE
Display Lab Journal in Collection View

### ISSUE THREAD

**Issue by noheton** [noheton/shepard#315]
***In GitLab by @wiglef on Nov 19, 2024, 11:41 GMT+1:***

# Feature Request

## :goal: Goal

As a frontend user I want to see all lab journal entries of data objects in my collection so that I have an overview.

## :medal: Acceptance Criteria

1. The Lab Journal content box is integrated displaying all lab journal entries of all data objects in the collection
   1. 1 Next to the title of the lab journal entry there is a link to the data object displayed. 
2. In the sidebar, the collection title is displayed under the "COLLECTION" Header and can be clicked to get back to the collection
3. If you have permissions to edit the lab journal in this collection, a edit and delete button is displayed

## :star: Special Testing hints

https://www.figma.com/design/ivBPq71AuUl2ohFnrzHIKF/Visual-Design?node-id=255-24857&m=dev

## :thinking: Risks & Open Questions

**Assignees:** xit-max-hey

*Migrated from GitLab: https://gitlab.com/dlr-shepard/shepard/-/work_items/315*

**Comment by noheton** [comment]
***In GitLab by @wiglef on Nov 19, 2024, 14:49 GMT+1:***

marked this issue as blocked by #293

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Nov 29, 2024, 16:03 GMT+1:***

We already did some of it with #293 For the ACs, we could adapt it to:
1. ~~There is a header with ID, Created at, Updated at Information~~
2. ~~There is a description content box - like in the data object view for data objects~~
3. ~~There is an attributes content box that shows key value pairs~~
4. The Lab Journal content box is integrated displaying all lab journal entries of data object in the collection
5. ~~All content boxes can be folded in and out and have a little counter to display how many elements are included~~
6. In the sidebar, the collection title is displayed under the "COLLECTION" Header and can be clicked to get back to the collection

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Dec 4, 2024, 13:22 GMT+1:***

marked this issue as blocked by #208

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Dec 13, 2024, 16:18 GMT+1:***

mentioned in commit f50c85920d41c74bd1c1d90cc38e6b27ef21497a

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Dec 16, 2024, 15:33 GMT+1:***

mentioned in commit a65394ab740e9518fc1d0e70bbdb842f2ba263c5

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Dec 16, 2024, 15:33 GMT+1:***

mentioned in commit b005c3e1d36fa4646bb70ac216f5d82a29e5e0e7

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Dec 16, 2024, 15:33 GMT+1:***

mentioned in commit fecbf7fa363b6a1c3d382fc40061cdc1babbd535

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Dec 16, 2024, 15:33 GMT+1:***

mentioned in commit 3270264aa11d3187ebd3e74db474042cb4f37bea

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Dec 16, 2024, 15:52 GMT+1:***

mentioned in commit 2b1d121407a4381d203e4d501cc03d2ceebd131d

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Dec 17, 2024, 11:36 GMT+1:***

mentioned in commit 6606949943d42e6f3c49887c7bed02c50e62cdc1

**Comment by noheton** [comment]
***In GitLab by @KlaraForner on Dec 17, 2024, 16:31 GMT+1:***

Testing: 
- arrow to expand / collapse should be disabled and not pointing up or down but to the right if entry is not expandable (eg because it's very short)

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Dec 17, 2024, 17:18 GMT+1:***

mentioned in commit cf0882f03124a71aed1c5abf3938c17650cf76c3

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Dec 17, 2024, 17:18 GMT+1:***

mentioned in commit 2839dfa62f4690bf9a8202a738b0f2a6d5be9e71

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Dec 17, 2024, 17:18 GMT+1:***

mentioned in commit 7d17a2132f4ba1f1434bb70c8eacd89f4947800b

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Dec 18, 2024, 10:17 GMT+1:***

mentioned in commit c345eaf55772ae25ec26fd79500abb3a99ed7a0a

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Dec 18, 2024, 10:19 GMT+1:***

We had (still have) the same issue for the LabJournal story. However, this issue was not part of the AC and requires an extra story. Since we need to discuss when a lab journal entry is too short.

I would therefore resolve this.

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Dec 19, 2024, 10:38 GMT+1:***

mentioned in commit d3cec5817d110e407d9221db2b39a2154d8a4da9

**Comment by noheton** [comment]
***In GitLab by @wiglef on Jan 31, 2025, 11:27 GMT+1:***

added dlr-shepard#16 as parent epic

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Feb 24, 2025, 17:48 GMT+1:***

marked this issue as related to #507
